### PR TITLE
Region box

### PR DESCRIPTION
**Pull request by shaobinspire**

### CODE REVIEW

File: Applications/Spire/Source/Ui/RegionBox.cpp [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+    m_tag_combo_box->setMinimumSize(minimumSize());
+  }
+  if(m_tag_combo_box->maximumSize() != maximumSize()) {
+    m_tag_combo_box->setMaximumSize(maximumSize());
+  }
+  QWidget::resizeEvent(event);
+}
+
+void RegionBox::on_current(const Region& region) {
+  while(m_tag_combo_box->get_current()->get_size() != 0) {
```
Comment (kranar) [inline Applications/Spire/Source/Ui/RegionBox.cpp:179]: This looks like a transaction.

File: Applications/Spire/Source/Ui/RegionBox.cpp [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+    m_tag_combo_box->get_current()->remove(
+      m_tag_combo_box->get_current()->get_size() - 1);
+  }
+  m_tag_combo_box->get_current()->push(region);
+}
+
+void RegionBox::on_submit(const std::shared_ptr<AnyListModel>& submission) {
+  auto region = Nexus::Region();
+  for(auto i = 0; i < submission->get_size(); ++i) {
+    region = region + std::any_cast<Region&&>(submission->get(i));
```
Comment (kranar) [inline Applications/Spire/Source/Ui/RegionBox.cpp:189]: Does ```+=``` work?

**Review by kranar — CHANGES_REQUESTED** [review]
Looks amazing.

File: Applications/Spire/Source/Ui/RegionBox.cpp (hunk 2 above)
Comment (shaobinspire) [reply Applications/Spire/Source/Ui/RegionBox.cpp:189]: No, 'Nexus::Region' does not define `+=` operation.

**Pull request closed without merging**